benchmark/CMakeLists.txt

```diff
@@ -87,16 +87,43 @@ endfunction(ginkgo_add_single_benchmark_executable)
 # All remaining arguments will be treated as source files
 function(ginkgo_add_typed_benchmark_executables name use_lib_linops)
     ginkgo_add_single_benchmark_executable(
-        "${name}" "${use_lib_linops}" "GKO_BENCHMARK_USE_DOUBLE_PRECISION" ${ARGN})
+        "${name}" "${use_lib_linops}" "GKO_BENCHMARK_USE_DOUBLE_PRECISION" "d" ${ARGN})
     ginkgo_add_single_benchmark_executable(
-        "${name}_single" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_PRECISION" ${ARGN})
+        "${name}_single" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_PRECISION" "s" ${ARGN})
     ginkgo_add_single_benchmark_executable(
-        "${name}_dcomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_DOUBLE_COMPLEX_PRECISION" ${ARGN})
+        "${name}_dcomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_DOUBLE_COMPLEX_PRECISION" "z" ${ARGN})
     ginkgo_add_single_benchmark_executable(
-        "${name}_scomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_COMPLEX_PRECISION" ${ARGN})
+        "${name}_scomplex" "${use_lib_linops}" "GKO_BENCHMARK_USE_SINGLE_COMPLEX_PRECISION" "c" ${ARGN})
 endfunction(ginkgo_add_typed_benchmark_executables)
 
 
+if (GINKGO_BUILD_CUDA)
+    enable_language(CUDA)
+    ginkgo_benchmark_cusp_linops(d GKO_BENCHMARK_USE_DOUBLE_PRECISION)
```

Comment:
I'm not too much of a fan of the current setup for multiprecision in general, I believe it should be shortened. But that's probably better in another PR.